core/tasks.py

```diff
@@ -207,7 +208,13 @@ def _score_submission(submission):
     return submission
 
 
-@shared_task(soft_time_limit=60 * 20, time_limit=60 * 21)
+@shared_task(
+    soft_time_limit=60 * 20,
+    time_limit=60 * 21,
+    autoretry_for=(SMTPServerDisconnected,),
+    default_retry_delay=10,
+    retry_kwargs={'max_retries': 5},
+)
 def score_submission(submission_id, dry_run=False, notify=False):
     submission = Submission.objects.get(pk=submission_id)
     if not dry_run:
```

Comment:
Idea: `dry_run` and `notify` should be exclusive, such that you can't trigger notifications on a dry run.

This would allow this to be split into 2 separate tasks: one for scoring, then optionally, one for notification. Then, if we have to retry a failed notification, we don't have to re-do all the work of scoring again.

What do you think?

Reply:
I agree, can we do it in a follow up PR?